### PR TITLE
*: update character-set-and-collation

### PR DESCRIPTION
**Pull request by zimulala**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Update execution results affect the number of rows.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs-cn/pull/8101/files

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by qiancai — APPROVED** [review]

File: mysql-compatibility.md [hunk 1]
```diff
@@ -125,6 +125,12 @@ TiDB 中的视图不可更新，不支持 `UPDATE`、`INSERT`、`DELETE` 等写
 
 详见 [TiDB 本地临时表与 MySQL 临时表的兼容性](/temporary-tables.md#与-mysql-临时表的兼容性)。
 
+### 字符集和排序规则
+
+* 关于字符集和排序规则的支持情况，详见 [字符集和排序规则](/character-set-and-collation.md)。
```
Comment (qiancai) [inline mysql-compatibility.md:130]:
```suggestion
* 关于 TiDB 对字符集和排序规则的支持情况，详见[字符集和排序规则](/character-set-and-collation.md)。
```

File: character-set-gbk.md [hunk 2]
```diff
@@ -25,7 +25,9 @@ SHOW COLLATION WHERE CHARSET = 'gbk';
 1 rows in set (0.00 sec)
 ```
 
-## 排序规则
+## MySQL 兼容性
+
+### 排序规则
```
Comment (qiancai) [inline character-set-gbk.md:30]:
```suggestion
### 排序规则兼容性
```

**Review by xiongjiwei — APPROVED** [review]

**Comment by zimulala** [comment]
PTAL @qiancai

File: character-set-gbk.md [hunk 3]
```diff
@@ -25,7 +25,9 @@ SHOW COLLATION WHERE CHARSET = 'gbk';
 1 rows in set (0.00 sec)
 ```
 
-## 排序规则
+## MySQL 兼容性
```
Comment (shichun-0415) [inline character-set-gbk.md:28]:
```suggestion
## 与 MySQL 的兼容性

本节介绍 TiDB 中启用 GBK 字符集与 MySQL 的兼容情况，包括排序规则和非法字符兼容性两方面。
```

**Review by shichun-0415 — APPROVED** [review]
rest LGTM

File: mysql-compatibility.md [hunk 4]
```diff
@@ -125,6 +125,12 @@ TiDB 中的视图不可更新，不支持 `UPDATE`、`INSERT`、`DELETE` 等写
 
 详见 [TiDB 本地临时表与 MySQL 临时表的兼容性](/temporary-tables.md#与-mysql-临时表的兼容性)。
 
+### 字符集和排序规则
+
+* 关于 TiDB 对字符集和排序规则的支持情况，详见[字符集和排序规则](/character-set-and-collation.md)。
+
+* 关于 GBK 字符集与 MySQL 的兼容情况，详见 [GBK 兼容情况](/character-set-gbk.md#mysql-兼容性)。
```
Comment (qiancai) [inline mysql-compatibility.md:132]:
```suggestion
* 关于 GBK 字符集与 MySQL 的兼容情况，详见 [GBK 兼容情况](/character-set-gbk.md#与-mysql-的兼容性)。
```

**Pull request merged**